File: backend/src/main/java/com/it342/backend/model/History.java
package com.it342.backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "history")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
public class History {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "report_id")
    private Report report;

    @ManyToOne
    @JoinColumn(name = "changed_by")
    private User changedBy;

    @Column(name = "old_status")
    private String oldStatus;

    @Column(name = "new_status")
    private String newStatus;

    private String notes;

    @Column(name = "changed_at")
    private LocalDateTime changedAt = LocalDateTime.now();
}

File: mobile/app/src/main/java/com/it342/teknoyfix/model/History.kt
package com.it342.teknoyfix.model

data class History(
    val id: Long = 0,
    val report: Report? = null,
    val changedBy: User? = null,
    val oldStatus: String? = null,
    val newStatus: String = "",
    val notes: String? = null,
    val changedAt: String? = null
)